corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js

```diff
@@ -128,7 +156,10 @@ hqDefine("cloudcare/js/formplayer/menus/controller", function () {
 
         var queryResponse = menuResponse.queryResponse;
         if (sidebarEnabled && menuResponse.type === "entities" && queryResponse)  {
-            var queryCollection = new Collection(queryResponse.displays);
+            let queryCollection = new Collection(queryResponse.displays);
+            if (queryResponse.hasOwnProperty("groupHeaders")) {
```

Comment:
I might have told you the wrong thing before. But `groupHeaders` will always be in the response. If there's no groups, the value will be `{}` so you'll need to also check if it's empty here and below

Reply:
replacing it with `Object.keys(queryResponse.groupHeaders).length > 0`